docs/website/contents/for-developers/HardWonWisdom.md

```diff
@@ -348,3 +348,77 @@ The key corresponding parts of the implementation are as follows.
 - The interpretation only depends on the eras' start and end bounds, not on the safe zone.
   The safe zone is instead used in an preceding step to safely determine the least possible value of the first unknown end bound.
   This determination is implemented by the `ouroboros-consensus:Ouroboros.Consensus.HardFork.Combinator.State.reconstructSummaryLedger` function.
+
+## The hard fork combinator interface
+
+To run the Consensus Layer on a block type `blk` requires the following (whether or not it's a `HardForkBlock`).
+
+- Values for various tracers, ChainDB options, etc, as required by `Ouroboros.Consensus.Node.run`.
+- A `ProtocolInfo blk` value, which is `TopLevelConfig blk` as well as initial state of the ledger and protocol.
+- Instances of `ConsensusProtocol (BlockProtocol blk)`, `LedgerSupportsProtocol blk`, and so on (see the `RunNode blk` class).
+
+Suppose there are two blocks types `F` and `G` for which the above is already available.
+To run the Consensus Layer instead on the hard fork combinator that sequences `F` into `G` requires the following additional information about `F` and `G`.
+
+- A value of `EraParams` for `F` and for `G` (ie `Shape [F, G]`), which decomposes as follows.
+    - The duration of every slot in the epoch as a [`NominalDiffTime`](https://hackage.haskell.org/package/time-1.12.2/docs/Data-Time-Clock.html#t:NominalDiffTime).
```

Comment:
In the epoch of `F` and `G`?

Reply:
One for each, yes; I'll update the text to be more explicit.

Reply:
Done.